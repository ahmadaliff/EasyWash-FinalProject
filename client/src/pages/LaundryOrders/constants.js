export const GET_ORDERS = 'laundryOrder/GET_ORDERS';
export const CHANGE_STATUS = 'laundryOrder/CHANGE_STATUS';
export const SET_ORDERS = 'laundryOrder/SET_ORDERS';
export const RESET_ORDERS = 'laundryOrder/RESET_ORDERS';
