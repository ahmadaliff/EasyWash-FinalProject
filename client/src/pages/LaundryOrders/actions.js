import { CHANGE_STATUS, GET_ORDERS, RESET_ORDERS, SET_ORDERS } from './constants';

export const actionGetOrders = (limit, page) => ({
  type: GET_ORDERS,
  limit,
  page,
});

export const actionSetOrders = (orders) => ({
  type: SET_ORDERS,
  orders,
});

export const actionChangeStatus = (id, newStatus, callback) => ({
  type: CHANGE_STATUS,
  id,
  newStatus,
  callback,
});

export const actionResetOrders = () => ({
  type: RESET_ORDERS,
});
