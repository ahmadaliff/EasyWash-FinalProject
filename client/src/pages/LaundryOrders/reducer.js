import _ from 'lodash';
import { produce } from 'immer';
import { CHANGE_STATUS, RESET_ORDERS, SET_ORDERS } from './constants';

export const initialState = {
  orders: null,
};

export const laundryOrdersReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case SET_ORDERS:
        draft.orders = action.orders;
        break;
      case CHANGE_STATUS: {
        const tempObj = { ...state.orders };

        const filteredDataOrders = tempObj.data.filter((val) => val.id !== action.id);
        const orderToChange = { ...tempObj.data.filter((val) => val.id === action.id)[0] };
        orderToChange.status = action.newStatus;
        filteredDataOrders.push(orderToChange);
        tempObj.data = _.sortBy(filteredDataOrders, [(val) => val.id]);

        draft.orders = tempObj;
        break;
      }
      case RESET_ORDERS:
        return initialState;
    }
  });

export default laundryOrdersReducer;
