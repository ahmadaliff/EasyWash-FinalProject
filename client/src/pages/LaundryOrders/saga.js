import { call, put, takeLatest } from 'redux-saga/effects';
import { setLoading, showPopup } from '@containers/App/actions';
import toast from 'react-hot-toast';
import { apiChangeStatus, apiGetLaundryOrders } from '@domain/api';
import intlHelper from '@utils/intlHelper';
import { actionSetOrders } from './actions';
import { CHANGE_STATUS, GET_ORDERS } from './constants';

function* sagaGetOrders({ limit, page }) {
  yield put(setLoading(true));
  try {
    const response = yield call(apiGetLaundryOrders, limit, page);
    yield put(actionSetOrders(response));
  } catch (error) {
    if (error?.response?.status === 400 || error?.response?.status === 404 || error?.response?.status === 401) {
      toast.error(intlHelper({ message: error.response.data.message }));
    } else {
      yield put(showPopup());
    }
  }
  yield put(setLoading(false));
}

function* sagaChangeStatus({ id, newStatus }) {
  yield put(setLoading(true));
  try {
    // eslint-disable-next-line object-shorthand
    const response = yield call(apiChangeStatus, { orderId: id, newStatus: newStatus });
    toast.success(intlHelper({ message: response.message }));
  } catch (error) {
    if (error?.response?.status === 400 || error?.response?.status === 404 || error?.response?.status === 401) {
      toast.error(intlHelper({ message: error.response.data.message }));
    } else {
      yield put(showPopup());
    }
  }
  yield put(setLoading(false));
}

export default function* laundryOrdersSaga() {
  yield takeLatest(GET_ORDERS, sagaGetOrders);
  yield takeLatest(CHANGE_STATUS, sagaChangeStatus);
}
