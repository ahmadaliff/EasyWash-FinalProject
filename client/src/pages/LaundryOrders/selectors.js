import { createSelector } from 'reselect';
import { initialState } from '@pages/LaundryServices/reducer';

const selectLaundryOrdersState = (state) => state.laundryOrders || initialState;

export const selectOrders = createSelector(selectLaundryOrdersState, (state) => state.orders);
