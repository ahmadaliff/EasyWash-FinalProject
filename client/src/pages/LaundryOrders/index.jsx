import PropTypes from 'prop-types';
import { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { connect, useDispatch } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { FormattedMessage, injectIntl } from 'react-intl';

import {
  Alert,
  Button,
  Card,
  IconButton,
  Stack,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TablePagination,
  TableRow,
} from '@mui/material';
import { ArrowBack, ArrowRight, DoNotDisturb, Done } from '@mui/icons-material';

import DetailOrder from '@components/DetailOrder';

import { selectOrders } from '@pages/LaundryOrders/selectors';
import { actionChangeStatus, actionGetOrders, actionResetOrders } from '@pages/LaundryOrders/actions';
import { actionAddChannel, actionDeleteChannel } from '@pages/ChatPage/actions';

import classes from '@pages/LaundryOrders/style.module.scss';

const LaundryOrders = ({ orders, intl: { formatMessage } }) => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(10);
  const [isDetailOpen, setIsDetailOpen] = useState(false);
  const [orderState, setOrderState] = useState(null);

  const columns = [
    { id: 'id', label: formatMessage({ id: 'app_order_id' }), maxWidth: 50 },
    { id: 'totalPrice', label: formatMessage({ id: 'app_total_price' }), minWidth: 100 },
    { id: 'status', label: 'status', minWidth: 100 },
  ];

  const formattedPrice = (val) =>
    new Intl.NumberFormat('id-ID', {
      style: 'currency',
      currency: 'IDR',
      minimumFractionDigits: 0,
    }).format(val);

  useEffect(() => {
    dispatch(actionGetOrders(rowsPerPage, page));
    return () => {
      if (orders) {
        dispatch(actionResetOrders());
      }
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [dispatch, page, rowsPerPage]);

  const buttonComp = (newStatus, row, idIntl, callback, nameClass = classes.buttonActionAcc) => (
    <Button
      type="button"
      className={nameClass}
      onClick={() => dispatch(actionChangeStatus(row.id, newStatus, callback))}
    >
      {nameClass === classes.buttonActionAcc ? <Done /> : <DoNotDisturb />}
      <div className={classes.email}>
        <FormattedMessage id={idIntl} />
      </div>
    </Button>
  );

  const actionComp = (order) => {
    switch (order.status) {
      case 'app_pending':
        return (
          <>
            {buttonComp('app_payment', order, 'app_acc', () => dispatch(actionAddChannel(order?.userId)))}
            {buttonComp('app_rejected', order, 'app_decline', undefined, classes.buttonActionDec)}
          </>
        );
      case 'app_pickUp':
        return buttonComp('app_onProcess', order, 'app_onProcess');
      case 'app_onProcess':
        return buttonComp('app_onDelivery', order, 'app_onDelivery');
      case 'app_onDelivery':
        return buttonComp('app_finish', order, 'app_finish', () => dispatch(actionDeleteChannel(order?.userId)));
      default:
        return 'No Action';
    }
  };

  const handleChangePage = (event, newPage) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event) => {
    setRowsPerPage(+event.target.value);
    setPage(0);
  };

  return (
    <main className={classes.mainWrap}>
      <div className={classes.header}>
        <div>
          <h3>
            <FormattedMessage id="app_orders_header" />
          </h3>
        </div>
        <Button
          variant="contained"
          className={classes.backButton}
          startIcon={<ArrowBack />}
          onClick={() => navigate(-1)}
        >
          <FormattedMessage id="app_back" />
        </Button>
      </div>
      <Card className={classes.card}>
        <TableContainer sx={{ maxHeight: 440 }}>
          <Table stickyHeader aria-label="sticky table">
            <TableHead>
              <TableRow>
                {columns.map((column) => (
                  <TableCell key={column.id} className={classes.tableHead}>
                    {column.label}
                  </TableCell>
                ))}
                <TableCell className={classes.tableHead}>
                  <FormattedMessage id="app_action" />
                </TableCell>
                <TableCell className={classes.tableHead}>Detail</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {orders?.data?.map((order) => (
                <TableRow hover key={order.id}>
                  {columns.map((column) => {
                    const value = order[column.id];
                    return (
                      <TableCell key={column.id} className={classes.tableBody}>
                        {
                          // eslint-disable-next-line no-nested-ternary
                          column.id === 'status'
                            ? formatMessage({ id: value })
                            : column.id === 'totalPrice'
                            ? formattedPrice(value)
                            : value
                        }
                      </TableCell>
                    );
                  })}

                  <TableCell className={classes.tableBody}>
                    <div className={classes.tableAction}>{actionComp(order)}</div>
                  </TableCell>
                  <TableCell>
                    <IconButton
                      onClick={() => {
                        setOrderState(order);
                        setIsDetailOpen(true);
                      }}
                    >
                      <ArrowRight />
                    </IconButton>
                  </TableCell>
                </TableRow>
              ))}
              <DetailOrder
                order={orderState}
                open={isDetailOpen}
                handleClose={() => {
                  setIsDetailOpen(false);
                }}
              />
              {orders?.data?.length === 0 && (
                <TableRow hover>
                  <TableCell colSpan={5}>
                    <Stack sx={{ width: '100%' }} spacing={2}>
                      <Alert severity="error">
                        <FormattedMessage id="app_404" />
                      </Alert>
                    </Stack>
                  </TableCell>
                </TableRow>
              )}
            </TableBody>
          </Table>
        </TableContainer>
        <TablePagination
          className={classes.tablePagination}
          rowsPerPageOptions={[10, 25, 100]}
          component="div"
          count={orders?.totalRows || 0}
          rowsPerPage={rowsPerPage}
          page={orders ? page : 0}
          onPageChange={handleChangePage}
          onRowsPerPageChange={handleChangeRowsPerPage}
        />
      </Card>
    </main>
  );
};

LaundryOrders.propTypes = {
  intl: PropTypes.object,
  orders: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  orders: selectOrders,
});

export default injectIntl(connect(mapStateToProps)(LaundryOrders));
