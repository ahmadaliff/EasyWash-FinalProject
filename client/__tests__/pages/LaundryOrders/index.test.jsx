import LaundryOrders from '@pages/LaundryOrders';
import { render, fireEvent, act } from '@utils/testHelper';

let wrapper;
const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

jest.mock('@pages/LaundryOrders/selectors', () => ({
  selectOrders: () => ({
    data: [
      {
        status: 'app_pickUp',
        userId: 1,
        totalRows: 1,
        id: 1,
        totalPrice: 20000,
      },
    ],
  }),
}));

jest.mock('@components/DetailOrder', () => jest.fn().mockImplementation(() => <p data-testid="dialog-detail-order" />));

beforeEach(() => {
  act(() => {
    wrapper = render(<LaundryOrders />);
  });
});

describe('Laundry Orders Page', () => {
  test('render', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('laundry-orders')).toBeInTheDocument();
  });
  test('navigate back when back button click', async () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('back-button');
    await act(async () => {
      fireEvent.click(button);
    });
    expect(mockNavigate).toHaveBeenCalledWith(-1);
  });
  test('open dialog when button click', async () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-dialog');
    act(() => {
      fireEvent.click(button);
    });
    expect(getByTestId('dialog-detail-order')).toBeInTheDocument();
  });
  test('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
