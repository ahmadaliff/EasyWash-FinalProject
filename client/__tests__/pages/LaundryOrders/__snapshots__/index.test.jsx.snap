// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Laundry Orders Page Matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="mainWrap"
        data-testid="laundry-orders"
      >
        <div
          class="header"
        >
          <div>
            <h3>
              Manajemen Pesanan
            </h3>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backButton css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="back-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowBackIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
            </span>
            Kembali
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiTableContainer-root css-41abqd-MuiTableContainer-root"
          >
            <table
              aria-label="sticky table"
              class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
            >
              <thead
                class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead css-y8ay40-MuiTableCell-root"
                    scope="col"
                  >
                    Id Pesanan
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead css-y8ay40-MuiTableCell-root"
                    scope="col"
                  >
                    Harga Total
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead css-y8ay40-MuiTableCell-root"
                    scope="col"
                  >
                    status
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead css-y8ay40-MuiTableCell-root"
                    scope="col"
                  >
                    Aksi
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead css-y8ay40-MuiTableCell-root"
                    scope="col"
                  >
                    Detail
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody css-1ex1afd-MuiTableCell-root"
                  >
                    1
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody css-1ex1afd-MuiTableCell-root"
                  >
                    Rp 20.000
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody css-1ex1afd-MuiTableCell-root"
                  >
                    Diambil
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody css-1ex1afd-MuiTableCell-root"
                  >
                    <div
                      class="tableAction"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium buttonActionAcc css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="DoneIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
                          />
                        </svg>
                        <div
                          class="email"
                        >
                          Diproses
                        </div>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      data-testid="button-dialog"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ArrowRightIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="m10 17 5-5-5-5v10z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="MuiTablePagination-root tablePagination css-jtlhu6-MuiTablePagination-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
            >
              <div
                class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
              />
              <p
                class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                id=":rd:"
              >
                Rows per page:
              </p>
              <div
                class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-16c50h-MuiInputBase-root-MuiTablePagination-select"
              >
                <div
                  aria-controls=":re:"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby=":rd: :rc:"
                  class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                  id=":rc:"
                  role="combobox"
                  tabindex="0"
                >
                  10
                </div>
                <input
                  aria-hidden="true"
                  aria-invalid="false"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name=":rf:"
                  tabindex="-1"
                  value="10"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <p
                class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
              >
                0–0 of 0
              </p>
              <div
                class="MuiTablePagination-actions"
              >
                <button
                  aria-label="Go to previous page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to previous page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowLeftIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Go to next page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to next page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowRightIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <p
            data-testid="dialog-detail-order"
          />
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="mainWrap"
      data-testid="laundry-orders"
    >
      <div
        class="header"
      >
        <div>
          <h3>
            Manajemen Pesanan
          </h3>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backButton css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="back-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowBackIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </span>
          Kembali
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiTableContainer-root css-41abqd-MuiTableContainer-root"
        >
          <table
            aria-label="sticky table"
            class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead css-y8ay40-MuiTableCell-root"
                  scope="col"
                >
                  Id Pesanan
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead css-y8ay40-MuiTableCell-root"
                  scope="col"
                >
                  Harga Total
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead css-y8ay40-MuiTableCell-root"
                  scope="col"
                >
                  status
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead css-y8ay40-MuiTableCell-root"
                  scope="col"
                >
                  Aksi
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead css-y8ay40-MuiTableCell-root"
                  scope="col"
                >
                  Detail
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody css-1ex1afd-MuiTableCell-root"
                >
                  1
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody css-1ex1afd-MuiTableCell-root"
                >
                  Rp 20.000
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody css-1ex1afd-MuiTableCell-root"
                >
                  Diambil
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody css-1ex1afd-MuiTableCell-root"
                >
                  <div
                    class="tableAction"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium buttonActionAcc css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="DoneIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
                        />
                      </svg>
                      <div
                        class="email"
                      >
                        Diproses
                      </div>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    data-testid="button-dialog"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="ArrowRightIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m10 17 5-5-5-5v10z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="MuiTablePagination-root tablePagination css-jtlhu6-MuiTablePagination-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
          >
            <div
              class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
            />
            <p
              class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
              id=":rd:"
            >
              Rows per page:
            </p>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-16c50h-MuiInputBase-root-MuiTablePagination-select"
            >
              <div
                aria-controls=":re:"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby=":rd: :rc:"
                class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                id=":rc:"
                role="combobox"
                tabindex="0"
              >
                10
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name=":rf:"
                tabindex="-1"
                value="10"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
            >
              0–0 of 0
            </p>
            <div
              class="MuiTablePagination-actions"
            >
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to previous page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowLeftIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </button>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to next page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <p
          data-testid="dialog-detail-order"
        />
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
